src/JustSaying/Messaging/Channels/IDownloadBuffer.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Amazon.SQS.Model;
+using JustSaying.AwsTools.MessageHandling;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Channels
+{
+    internal interface IDownloadBuffer
```

Comment:
This isn't very constructive, but something bugs me about the name `DownloadBuffer`, is there something more descriptive like `MessageReceiverBuffer`?

Reply:
Yeah agree here, we went around a bit on the naming here - we'll have another think. `IMessageReceiveBuffer` sounds like a good option...